### PR TITLE
feat(#3843): Add support for CanvasState.reset

### PR DESCRIPTION
**Pull request by Nul-led**
- Resolves issue #3843
- Documented method according to https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/reset

### CODE REVIEW

**Comment by Nul-led** [comment]
Not exactly sure if i need to put a notice into the changelog. Ill leave it at this for now.

**Comment by Nul-led** [comment]
Ok with that, its ready for review now i think

**Review by daxpedda — CHANGES_REQUESTED** [review]
Yes, this needs an entry in the changelog please.

**Comment by Nul-led** [comment]
@daxpedda good enough?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,7 @@
 * Add bindings for `RTCPeerConnectionIceErrorEvent`.
   [#3835](https://github.com/rustwasm/wasm-bindgen/pull/3835)
 
-* Add bindings for `CanvasState.reset()`, affecting `CanvasRenderingContext2D` and `OffscreenCanvasRenderingContext2D`.
-
-[#3844](https://github.com/rustwasm/wasm-bindgen/pull/3844)
+* Add bindings for `CanvasState.reset()`, affecting `CanvasRenderingContext2D` and `OffscreenCanvasRenderingContext2D`. [#3844](https://github.com/rustwasm/wasm-bindgen/pull/3844)
```
Comment (daxpedda) [inline CHANGELOG.md:11]:
```suggestion
* Add bindings for `CanvasState.reset()`, affecting `CanvasRenderingContext2D` and `OffscreenCanvasRenderingContext2D`.
  [#3844](https://github.com/rustwasm/wasm-bindgen/pull/3844)
```
Keeping in line with style so far.

**Review by daxpedda — APPROVED** [review]
Thanks!

**Pull request merged**